sympy/crypto/crypto.py

```diff
@@ -3324,3 +3324,84 @@ def decipher_bg(message, key):
         orig_msg += (m ^ b)
 
     return orig_msg
+
+
+########### XOR Cipher ############
+
+def encipher_xor(message, key):
+    """
+    Performs XOR encryption on plaintext and returns ciphertext.
+
+    Examples
+    ========
+
+    >>> from sympy.crypto.crypto import encipher_xor
+    >>> encipher_xor("THISISASTRING", "ABCD")
+    '150a0a170811021715100a0a06'
+
+    Parameters
+    ==========
+
+    message: string, the message to be encrypted.
+    key: string, the key to be used for encryption of the message.
+
+    Returns
+    =======
+    The encrypted ciphertext.
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/XOR_cipher
+
+    """
+
+    encrypted_text = ""
+    itr = 0
+    for f in range(len(message)):
+        encrypted_text += hex(ord(key[itr]) ^ ord(message[f]))[2:].zfill(2)
+        itr += 1
+        if len(key) <= itr:
+            itr = 0
+    return encrypted_text
+
+
+def decipher_xor(ciphertext, key):
+    """
+    Performs XOR decryption on ciphertext and returns plaintext.
+
+    Examples
+    ========
+
+    >>> from sympy.crypto.crypto import decipher_xor
+    >>> decipher_xor("150a0a170811021715100a0a06", "ABCD")
+    'THISISASTRING'
+
+    Parameters
+    ==========
+
+    ciphertext: requires a hexadecimal number
+    key: string, the key to be used for decryption of the message.
+
+    Returns
+    =======
+    The decrypted plaintext.
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/XOR_cipher
+
+    """
+    decrypted_text = ""
+    itr = 0
+    val_in_uni = ""
+    for i in range(0, len(ciphertext), 2):
+        val_in_uni += bytes.fromhex(ciphertext[i:i+2]).decode('utf-8')
```

Comment:
This doesn't work in Python 2 probably. Though we are removing Python 2 from the tests, wait for #18243

Reply:
Yeah, I was about to say that. :)